src/utils.ts

```diff
@@ -5,14 +5,24 @@ interface InjectCodeOptions {
     styleId?: string;
 }
 
-export type InjectCode = (cssCode: string, options: InjectCodeOptions) => string;
+export type InjectCode = (cssCode: string, options: InjectCodeOptions) => void;
 
 const cssInjectedByJsId = '\0vite/all-css';
 
-const defaultInjectCode: InjectCode = (cssCode, { styleId }) =>
-    `try{if(typeof document != 'undefined'){var elementStyle = document.createElement('style');${
-        typeof styleId == 'string' && styleId.length > 0 ? `elementStyle.id = '${styleId}';` : ''
-    }elementStyle.appendChild(document.createTextNode(${cssCode}));document.head.appendChild(elementStyle);}}catch(e){console.error('vite-plugin-css-injected-by-js', e);}`;
+function injectStyleTag(css: string, options: InjectCodeOptions) {
+    try {
+        if (typeof document != 'undefined') {
+            const $style = document.createElement('style');
+            if (options.styleId) {
+                $style.id = options.styleId;
+            }
+            $style.appendChild(document.createTextNode(css));
+            document.head.appendChild($style);
+        }
+    } catch (e) {
+        console.error('vite-plugin-css-injected-by-js', e);
+    }
+}
```

Comment:
We need to let it be a generated string, as this way we can avoid unnecessary code being added and several ifs executed

Reply:
Just to confirm , the difference in code here is that this code: 
``` javascript           
if (options.styleId) {
   $style.id = options.styleId;
}
```

will always be in the code added, even if someone does not provide a `styleId`.  Is my understanding correct that you would like to have no branching within the runtime code?

Reply:
Exactly by using the function we always add checks, in that case is for styleId, in runtime and I prefer to not add needless code into the output